src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -42,11 +42,32 @@ public static object ConvertFromJson(string input, out ErrorRecord error)
             {
                 throw new ArgumentNullException("input");
             }
+
             error = null;
 #if CORECLR
             object obj = null;
             try
             {
+                // JsonConvert.DeserializeObject does not throw an exception when an invalid array is passed,
+                // so we need to do extra logic to identify this scenario using JArray.Parse(input).
+                if ((input.IndexOf("[", StringComparison.OrdinalIgnoreCase) >= 0) ||
```

Comment:
No, this should be an '||' because we are trying to detect an invalid array. These are some examples:
1) '["a"'
2) '['
3) '"a"]'